src/app/ui/shared/appoitment-form/appoitment-form.component.ts

```diff
@@ -0,0 +1,149 @@
+import { DoctorAppointmentsService } from './../../../services/doc-appointment-service/doctor-appointments.service';
+import { DoctorsAppointmentDTO } from 'src/app/data/modelDTO/doctors-appointment-dto';
+import { UserService } from './../../../services/user/user.service';
+import { DoctorServiceDTO } from './../../../data/modelDTO/dorctor-service-DTO';
+import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { debounceTime, distinctUntilChanged, take } from 'rxjs/operators';
+import { DoctorDTO } from 'src/app/data/modelDTO/doctor-DTO';
+import { DoctorService } from 'src/app/services/doctor/doctor.service';
+import { DoctorServicesService } from 'src/app/services/doctor-service/doctor-services.service';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { UserDto } from 'src/app/data/modelDTO/user-dto';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-appoitment-form',
+  templateUrl: './appoitment-form.component.html',
+  styleUrls: ['./appoitment-form.component.scss'],
+})
+export class AppoitmentFormComponent implements OnInit {
+  sampleForm = new FormGroup({
+    startDate: new FormControl(null, Validators.required),
+    startTime: new FormControl(null, Validators.required),
+    medicId: new FormControl(null, Validators.required),
+    medicName: new FormControl(null, Validators.required),
+    medService: new FormControl(null, Validators.required),
+    medLocation: new FormControl(null, Validators.required),
+    pacientName: new FormControl('', Validators.required),
+    pacientId: new FormControl('', Validators.required),
+    pacientAnimal: new FormControl({}, Validators.required),
+  });
+  medicServices: string[];
+  formTitle: string;
+  pacientName: string;
+  users: Observable<any>;
+  animals: AnimalUtilInfo[];
+  doctorAppointment: DoctorsAppointmentDTO;
+  appoinementFormPlaceHolder;
+  @ViewChild('pacientList') pacientList: any;
+  @ViewChild('inputPacientName') inputPacientName: any;
+
+  constructor(
+    private doctorService: DoctorService,
+    private servicesService: DoctorServicesService,
+    private activeModal: NgbActiveModal,
+    private userService: UserService,
+    private doctorAppointmentService: DoctorAppointmentsService,
+    private appointmentFormService: AppointmentFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.appoinementFormPlaceHolder = appointmentFormData;
+    this.formTitle = appointmentFormData.title;
+    this.pacientName = '';
+
+    this.doctorService
+      .getDoctorById('o2Jt7YS9zCWvBfDWY08X')
+      .pipe(take(1))
+      .subscribe((medic: DoctorDTO) => {
+        this.sampleForm.patchValue({
+          medicName: medic.doctorName,
+          medLocation: medic.location,
+          medicId: 'o2Jt7YS9zCWvBfDWY08X',
+        });
+      });
+
+    this.servicesService
```

Comment:
also this , you can use it in doctorService and chian the 2 observables together - if you have questions you have my number:D